x-pack/legacy/plugins/apm/public/components/shared/charts/Histogram/index.js

```diff
@@ -120,11 +124,25 @@ export class HistogramInner extends PureComponent {
       .range([XY_HEIGHT, 0])
       .nice();
 
+    const [xMinZone, xMaxZone] = [xMin, xMax].map(x => {
+      return x - getTimezoneOffsetInMs(x);
+    });
+
+    const xTickValues = isTimeSeries
+      ? d3.time.scale
+          .utc()
+          .domain([xMinZone, xMaxZone])
+          .range([0, XY_WIDTH])
+          .ticks(X_TICK_TOTAL)
+          .map(x => {
+            const time = x.getTime();
+            return new Date(time + getTimezoneOffsetInMs(time));
+          })
+      : undefined;
```

Comment:
Maybe a helper? `getNiceTimeTickValues()`, something like that? We can re-use it for `plotUtils` and it would have to added benefit of making clear what this piece of code does.